storage/src/desktop/storage_reference_desktop.cc

```diff
@@ -404,10 +409,12 @@ Future<Metadata> StorageReferenceInternal::PutBytes(
 
 Future<Metadata> StorageReferenceInternal::PutBytesInternal(
     const void* buffer, size_t buffer_size, Listener* listener,
-    Controller* controller_out) {
+    Controller* controller_out, const char* content_type) {
   auto* future_api = future();
   auto handle = future_api->SafeAlloc<Metadata>(kStorageReferenceFnPutBytes);
-  auto send_request_funct{[&, buffer, buffer_size, listener,
+
+  std::string content_type_str = content_type ? content_type : "";
```

Comment:
What's the significance of turning this into a std::string if it is going to be turned back into a c_str before it is passed into PrepareRequest?
(Same question for content_type_str in PutFileInternal)
I suppose this might keep the string in scope for the duration of retrying, but not certain.

Reply:
It's copying the data into a local variable - because this is an asynchronous operation, we can't rely on the string that was passed into content_type being around forever. So we make a local copy for the duration of the retries.

Reply:
(To be completely clear, adding it to the lambda capture list is what causes it to be COPIED into the other thread.)